### PR TITLE
fix: adjust GetPaymentsLimit to work correctly with historical blocks, adjust sb params on regtest, tweak tests

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
Noticed a couple of things while I was trying to figure out if an [issue](https://github.com/dashpay/dash/pull/5627#discussion_r1367153099) @knst mentioned in #5627 could actually exist:
1. `GetPaymentsLimit()` won't work correctly with historical blocks rn. We don't use it that way internally but it could be done via rpc and it should provide correct results.
2. superblock params on regtest are too small to test them properly
3. because of (2) and a huge v20 activation window (comparing to sb params) `feature_governance.py` doesn't test v20 switching states. There's also no "sb on v20 activation block" test.

~NOTE: based on #5639 atm~

## What was done?
fix it, pls see individual commits

## How Has This Been Tested?
run tests

## Breaking Changes
n/a

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by ogabrielides — DISMISSED** [review]
ACK 👍

**Comment by UdjinM6** [comment]
rebased on develop and added efce411b8ddc60e8787170b7790900fdf7c09b80

**Review by ogabrielides — APPROVED** [review]
utACK

File: src/governance/classes.cpp [hunk 1]
```diff
@@ -494,14 +494,16 @@ CAmount CSuperblock::GetPaymentsLimit(int nBlockHeight)
         return 0;
     }
 
-    const CBlockIndex* tipIndex = ::ChainActive().Tip();
-    const auto v20_state = llmq::utils::GetV20State(tipIndex);
+    const CBlockIndex* pindex = ::ChainActive().Tip();
```
Comment (PastaPastaPasta) [inline src/governance/classes.cpp:497]: nit: could make this gsl::not_null

File: test/functional/feature_governance.py [hunk 2]
```diff
@@ -52,131 +91,148 @@ def run_test(self):
...
 
         assert_equal(len(self.nodes[0].gobject("list-prepared")), 3)
         assert_equal(len(self.nodes[0].gobject("list")), 0)
 
-        p0_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p0_collateral_prepare["hex"], p0_collateral_prepare["collateralHash"])
-        p1_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p1_collateral_prepare["hex"], p1_collateral_prepare["collateralHash"])
-        p2_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p2_collateral_prepare["hex"], p2_collateral_prepare["collateralHash"])
+        self.p0_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p0_collateral_prepare["hex"], p0_collateral_prepare["collateralHash"])
+        self.p1_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p1_collateral_prepare["hex"], p1_collateral_prepare["collateralHash"])
+        self.p2_hash = self.nodes[0].gobject("submit", "0", 1, proposal_time, p2_collateral_prepare["hex"], p2_collateral_prepare["collateralHash"])
```
Comment (PastaPastaPasta) [inline test/functional/feature_governance.py:147]: nit: would have been nice if these "self." only changes were in their own commit for easier review

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge; a few comments but nothing blocking

**Pull request merged**